hetmech/hetmat.py

```diff
@@ -178,6 +203,57 @@ def permutations(self):
             permutations[name] = permutation
         return permutations
 
+    def permute_graph(self, num_new_permutations=None, namer=None, start_from=None,
+                      multiplier=10, excluded_pair_set=set(), seed=0, log=True):
+        """
+        Generate and save permutations of the HetMat adjacency matrices.
+
+        Parameters
+        ----------
+        num_new_permutations : int
+        namer : generator
+        start_from : str
+        multiplier : int
+        excluded_pair_set : set
+        seed : int
+        log : bool
+        """
+        if namer is None:
+            # If no namer given, continue increasing names by one for new permutations
+            def namer():
+                i = 1
+                while i:
+                    yield str(i).zfill(3)
+                    i += 1
+            namer = namer()
+        for _ in range(num_new_permutations):
+            permutation_name = next(namer)
+            name = f'{permutation_name}.hetmat/'
+            new_permutation_path = self.permutations_directory.joinpath(name)
+
+            new_hetmat = HetMat(new_permutation_path, initialize=True)
+            new_hetmat.is_permutation = True
+            new_hetmat.metagraph_path.symlink_to('../../metagraph.json')
+            new_hetmat.nodes_directory.rmdir()
+            new_hetmat.nodes_directory.symlink_to('../../nodes', target_is_directory=True)
+
+            if start_from is not None:
+                original_hetmat = self.permutations[start_from]
+            else:
+                original_hetmat = self
+
+            metaedges = list(self.metagraph.get_edges(exclude_inverts=True))
+            for metaedge in metaedges:
+                rows, cols, original_matrix = original_hetmat.metaedge_to_adjacency_matrix(
+                    metaedge, dense_threshold=1)
+                is_directed = (metaedge.direction != 'both')
+                permuted_matrix, stats = _permute_matrix(
+                    original_matrix, directed=is_directed, multiplier=multiplier,
+                    excluded_pair_set=excluded_pair_set, seed=seed, log=log)
+                path = new_hetmat.get_edges_path(metaedge, file_format='sparse.npz')
+                scipy.sparse.save_npz(str(path), permuted_matrix, compressed=True)
```

Comment:
Could there be a potential issue where `_permute_matrix` doesn't return something sparse?

Reply:
Isn't this converting the permutated matrix to the type of the input?

```diff
+    # Keep the same sparse type as adjacency_matrix
+    permuted_adjacency = type(adjacency_matrix)(permuted_adjacency)
```

Reply:
That only allows the interconversion between different types of sparse matrices. Do we want to be able to save them either way?

Reply:
Not necessarily, but will this break if someone passes a dense array to the function?

Reply:
Just added dense support